### PR TITLE
test: Cover ingredientBrandScreens decoding edge cases and Analysis badge forward-compat

### PR DESCRIPTION
**Pull request by qnaveed87**
## Pull Request Description

Adds iOS-side test coverage for PP-3506 — verifying the SDK behaves correctly
across the four `ingredientBrandScreens` decoding cases: valid array, unknown
values, and empty. No production code changes; every case works
against Swift's auto-synthesized `Codable` decoder plus the pre-existing
`AnalysisViewController.addPoweredByGiniBadgeIfEnabled` filter.

[PP-3506](https://ginis.atlassian.net/browse/PP-3506)

### What ships

- **`ClientConfigurationTests` (Swift Testing):**
  - Merged R1+R4 into `ingredientBrandScreensDecodesArrayVerbatimIncludingUnknownScreenNames`
    — widened array to `["Analysis", "Foo", "UNKNOWN"]`, proves unknown screen
    names decode verbatim.
  - New `throwsTypeMismatchWhenIngredientBrandScreensIsNotAnArray` — pins loud
    failure on a non-array value.
  - New `decodingFailsWhenIngredientBrandScreensKeyIsAbsent` — pins the
    strict-loud contract explicitly.
- **`clientConfigurationWithIngredientBrand.json`** restructured into a
  three-variant envelope (`valid`, `malformed`, `missing`); a small
  `ingredientBrandFixture(variant:)` helper extracts each sub-object.
- **`AnalysisViewControllerTests` (XCTest, matches neighbouring file style):**
  - `testAnalysisShowsBadgeWhenScreensListContainsAnalysisMixedWithUnknownValues`
  - `testAnalysisHidesBadgeWhenScreensListContainsOnlyUnknownValues`
- **`specs/PP-3506-feature.md`** — full spec + scope decision + implementation ledger.

### Public API impact

**None.** No `Sources/**` files touched.

## Notes for Reviewers

- **Simulator:** iPhone 17 / iOS 26.2 (CI parity per `.github/workflows/shared-config.yml`).
- **Xcode:** 26.2.
- **Verified locally:**
  - `xcodebuild test -scheme GiniBankAPILibrary` → 53/53 passed.
  - `xcodebuild test -scheme GiniCaptureSDK` → 191/191 passed.
  - Full suites, not just the touched files.

[PP-3506]: https://ginis.atlassian.net/browse/PP-3506?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PP-2572]: https://ginis.atlassian.net/browse/PP-2572?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BankAPILibrary/GiniBankAPILibrary/Tests/GiniBankAPILibraryTests/ClientConfigurationTests.swift [hunk 1]
```diff
@@ -229,34 +229,26 @@ struct ClientConfigurationTests {
...
+
+        let error = #expect(throws: DecodingError.self) {
+            try JSONDecoder().decode(ClientConfiguration.self, from: data)
         }
-        """.data(using: .utf8)!
+
+        guard case .typeMismatch? = error else {
+            Issue.record("Expected DecodingError.typeMismatch for `ingredientBrandScreens`, got \(String(describing: error))")
+            return
+        }
```
Comment (zladzeyka) [inline BankAPILibrary/GiniBankAPILibrary/Tests/GiniBankAPILibraryTests/ClientConfigurationTests.swift:243]: Request: Let's follow the same test flow and add 
`#expect(context.codingPath.last?.stringValue == "ingredientBrandScreens",
  ▎                 "Expected type mismatch on ingredientBrandScreens, got path \(context.codingPath)")`

**Review by zladzeyka — CHANGES_REQUESTED** [review]
Thanks for the PR

File: BankAPILibrary/GiniBankAPILibrary/Tests/GiniBankAPILibraryTests/ClientConfigurationTests.swift (hunk 1 above)
Comment (qnaveed87) [reply BankAPILibrary/GiniBankAPILibrary/Tests/GiniBankAPILibraryTests/ClientConfigurationTests.swift:243]: Done

**Review by zladzeyka — APPROVED** [review]
